gulpfile.babel.js

```diff
@@ -77,74 +79,69 @@ const bs = browserSync.create(),
-          .pipe(errorNotifier())
+    .pipe(errorNotifier())
 
-          // Inline assets
-          .pipe(inline(OPTIONS.inline))
+    // Inline assets
+    .pipe(inline(OPTIONS.inline))
 
-          // JS
-          .pipe(scripts.extract('script'))
-            .pipe(eslint())
-            .pipe(eslint.format())
-            .pipe(gulpif(!argv.debug, eslint.failAfterError()))
-            .pipe(rollup(OPTIONS.rollup))
-          .pipe(scripts.restore())
+    // JS
+    .pipe(scripts.extract('script'))
+      .pipe(processJs())
+    .pipe(scripts.restore())
 
-          // CSS
-          .pipe(styles.extract('style'))
-            .pipe(postcss(OPTIONS.postcss))
-          .pipe(styles.restore())
+    // CSS
+    .pipe(styles.extract('style'))
+      .pipe(postcss(OPTIONS.postcss))
+    .pipe(styles.restore())
 
-          .pipe(gulpif(!argv.debug, minify(OPTIONS.HTMLmin)))
+    .pipe(gulpif(!argv.debug, minify(OPTIONS.HTMLmin)))
 
-          .pipe(size({ gzip: true }))
-        .pipe(gulp.dest('.'))
+    .pipe(size({ gzip: true }))
+  .pipe(gulp.dest('.'))
 });
 
-gulp.task('build:tests:html', () => {
-  let scripts = processInline();
-
-  return gulp.src(['test/**/*.html'])
-          .pipe(errorNotifier())
+gulp.task('build:tests', () => {
+  const js = filter((file) => /\.(js)$/.test(file.path), { restore: true }),
+        html = filter((file) => /\.(html)$/.test(file.path), { restore: true }),
+        scripts = processInline();
 
-          // Inline assets
-          .pipe(inline(OPTIONS.inline))
+  return gulp.src(['test/**/*'])
+    .pipe(errorNotifier())
 
-          // JS
-          .pipe(scripts.extract('script'))
-            .pipe(eslint())
-            .pipe(eslint.format())
-            .pipe(rollup(OPTIONS.rollup))
-          .pipe(scripts.restore())
+    .pipe(html)
+      .pipe(inline(OPTIONS.inline))
+      .pipe(scripts.extract('script'))
+        .pipe(processJs())
+      .pipe(scripts.restore())
+    .pipe(html.restore)
 
-        .pipe(gulp.dest('./.test'));
-});
+    .pipe(js)
+      .pipe(processJs())
+    .pipe(js.restore)
 
-gulp.task('build:tests:js', () => {
-  return gulp.src(['test/*/*.js'])
-          .pipe(errorNotifier())
-          .pipe(eslint())
-          .pipe(eslint.format())
-          .pipe(rollup(OPTIONS.rollup))
-          .pipe(gulp.dest('./.test'));
+  .pipe(gulp.dest('.test'));
 });
 
 wct.gulp.init(gulp);
 
-gulp.task('serve', (callback) => bs.init(OPTIONS.browserSync));
-gulp.task('refresh', () => bs.reload());
-
-gulp.task('build:tests', ['build:tests:html', 'build:tests:js']);
-
-gulp.task('test', ['build', 'build:tests', 'test:local']);
+gulp.task('test', () => gulprun('build:tests', 'test:local'));
 
 gulp.task('watch:src', () => gulp.watch(['src/**/*'], () => gulprun('build', 'refresh')));
-gulp.task('watch:tests', () => gulp.watch(['src/**/*', 'test/**/*'], () => gulprun('build:tests')))
+gulp.task('watch:tests', () => gulp.watch(['test/**/*'], () => gulprun('build:tests')))
```

Comment:
This needs to watch the `src` as the `tests` pull in exported functions, see the `viewable` unit tests.